### DISCUSSION TITLE
R link virgin mode

### DISCUSSION THREAD

**Discussion by Alecsbdj** [cedricp/ddt4all#1146] [Q&A]
Howdy! I made a small mistake, I own a Renault Talisman with R-Link2. I have virginized it, but when I try to load it with pyrene macro from the wine, I can't bring it back to normal. the settings hardly come back at all. what can i do in this case? thank you!
version r link: 2.2.17.803

**Comment by AlMeu** [comment]
@Alecsbdj I think your unit is too old at base and that perhaps it is the end of it life.
If so, take a look for a new unit for change.

**Comment by Furtif** [comment]
read https://github.com/cedricp/ddt4all/discussions/1138

**Comment by shrlnm** [comment]
@Alecsbdj Please share the log of your try to implement the macro

**Reply by Alecsbdj** [reply]
[
[ecu_log.txt](https://github.com/user-attachments/files/15975430/ecu_log.txt)
[elm_log.txt](https://github.com/user-attachments/files/15975431/elm_log.txt)
](url)

**Reply by Alecsbdj** [reply]
Error file NO DATA 346

**Reply by shrlnm** [reply]
I suppose that "NO DATA" in this case related to the problem inside ELM as so there are too long timeouts 


>[19:56:16.760]270000E600320000
<[**0.696**]270000E600320000
300100FFFFFFFFFF

Is it BT or WIFI?
